### ISSUE TITLE
gem_source option in gem module doesn't appear to work properly

### ISSUE THREAD

**Issue by vadimberezniker** [ansible/ansible#2314]
The gem module appears to have the following code which overwrites gem_source with name. I am having trouble understanding what the intention was.

``` python
if module.params['gem_source'] is not 'null'
   module.params['gem_source'] = module.params['name']
```

I changed it locally to

``` python
-    if module.params['gem_source'] is not 'null':
+    if not module.params['gem_source']:
         module.params['gem_source'] = module.params['name']
```

and that appears to do the right thing.

**Comment by dhozac** [comment]
Looks good. Would you like to submit a pull request for that?

**Comment by dhozac** [comment]
Fixed by #2323.
